rpc/http.go

```diff
@@ -195,6 +199,50 @@ func newCorsHandler(srv *Server, allowedOrigins []string) http.Handler {
 		AllowedMethods: []string{http.MethodPost, http.MethodGet},
 		MaxAge:         600,
 		AllowedHeaders: []string{"*"},
+		Debug:          true,
 	})
 	return c.Handler(srv)
 }
+
+// hostHandler is a handler which validates the Host-header of incoming requests.
+// The hostHandler can prevent DNS rebinding attacks, which do not utilize CORS-headers,
+// since they do in-domain requests against the RPC api. Instead, we can see on the Host-header
+// which domain was used, and validate that against a whitelist.
+type hostHandler struct {
+	AllowedHosts []string
```

Comment:
Instead of storing a slice, lets store a `map[string]struct{}` instead. That way we can do quick lookups without incurring a linear scanning cost for every single request.